common/environment/environment.cpp

```diff
@@ -1732,6 +1822,51 @@ unsigned CConstDropZoneIteratorByComputer::count() const
 
 //--------------------------------------------------
 
+
+CConstDropZoneServerInfoIterator::CConstDropZoneServerInfoIterator(const IConstDropZoneInfo * dropZone)
+{
+    Owned<IEnvironmentFactory> factory = getEnvironmentFactory();
+    constEnv.setown((CLocalEnvironment *)factory->openEnvironment());
+    serverListIt.setown(dropZone->getPTree().getElements("ServerList"));
+    maxIndex = dropZone->getPTree().getCount("ServerList");
+}
+
+bool CConstDropZoneServerInfoIterator::first()
+{
+    bool hasFirst = serverListIt->first();
+    if (hasFirst)
+        curr = new CConstDropZoneServerInfo(constEnv, &serverListIt->query());
```

Comment:
need to be careful with existing 'curr' , e.g. if first() recalled.

It should release curr.
Probably easiest if you make curr a Owned and setown here.